Index: feeds/packages/utils/avrdude/Makefile
===================================================================
--- feeds/packages/utils/avrdude/Makefile	(révision 34199)
+++ feeds/packages/utils/avrdude/Makefile	(copie de travail)
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=avrdude
-PKG_VERSION:=5.8
+PKG_VERSION:=5.11.1
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://download.savannah.gnu.org/releases/avrdude
-PKG_MD5SUM:=d459dcf47fa7fc2a29a70d68ebff4886
+PKG_MD5SUM:=3a43e288cb32916703b6945e3f260df9
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -22,7 +22,7 @@
   CATEGORY:=Utilities
   TITLE:=AVR Downloader/UploaDEr
   URL:=http://www.bsdhome.com/avrdude/
-  DEPENDS:=+libncurses +libreadline +libusb
+  DEPENDS:=+libusb
 endef
 
 define Package/avrdude/description
