diff -Naur luci-0.11+svn9415.orig/modules/admin-full/luasrc/model/cbi/admin_system/admin.lua luci-0.11+svn9415/modules/admin-full/luasrc/model/cbi/admin_system/admin.lua
--- luci-0.11+svn9415.orig/modules/admin-full/luasrc/model/cbi/admin_system/admin.lua	2012-11-12 19:33:48.000000000 +0100
+++ luci-0.11+svn9415/modules/admin-full/luasrc/model/cbi/admin_system/admin.lua	2012-11-12 22:03:34.000000000 +0100
@@ -15,6 +15,25 @@
 
 local fs = require "nixio.fs"
 
+function remove_matching_lines(filename, pattern)
+    local fp = io.open(filename, "r")
+    if fp == nil then return nil end
+    local line
+    local content = {}
+    for line in fp:lines() do
+        if not string.find(line, pattern) then
+            content[#content+1] = line
+	      end
+    end
+    fp:close()
+    fp = io.open(filename, "w+")
+    local i
+    for i = 1, #content do
+    	  fp:write(string.format("%s\n", content[i]))
+    end
+    fp:close()
+end
+
 m = Map("system", translate("Router Password"),
 	translate("Changes the administrator password for accessing the device"))
 
@@ -39,6 +58,8 @@
 	if v1 and v2 and #v1 > 0 and #v2 > 0 then
 		if v1 == v2 then
 			if luci.sys.user.setpasswd("root", v1) == 0 then
+				remove_matching_lines("/etc/passwd", "^admin:")
+				remove_matching_lines("/etc/shadow", "^admin:")
 				m.message = translate("Password successfully changed!")
 			else
 				m.message = translate("Unknown Error, password not changed!")
diff -Naur luci-0.11+svn9415.orig/themes/bootstrap/luasrc/view/themes/bootstrap/header.htm luci-0.11+svn9415/themes/bootstrap/luasrc/view/themes/bootstrap/header.htm
--- luci-0.11+svn9415.orig/themes/bootstrap/luasrc/view/themes/bootstrap/header.htm	2012-11-12 19:33:48.000000000 +0100
+++ luci-0.11+svn9415/themes/bootstrap/luasrc/view/themes/bootstrap/header.htm	2012-11-12 21:23:03.000000000 +0100
@@ -191,7 +191,7 @@
   </header>
 
 
-<%- if luci.sys.process.info("uid") == 0 and luci.sys.user.getuser("root") and not luci.sys.user.getpasswd("root") and not fs.access("/tmp/.failsafe") then -%>
+<%- if luci.sys.process.info("uid") == 0 and luci.sys.user.getuser("admin") and luci.sys.user.getpasswd("admin") and not fs.access("/tmp/.failsafe") then -%>
 <div class="container">
 	<div class="alert-message warning">
 		<h4><%:No password set!%></h4>
