Index: target/linux/ar71xx/Makefile
===================================================================
--- target/linux/ar71xx/Makefile	(revision 34322)
+++ target/linux/ar71xx/Makefile	(working copy)
@@ -11,7 +11,7 @@
 BOARDNAME:=Atheros AR7xxx/AR9xxx
 FEATURES:=squashfs jffs2 targz
 CFLAGS:=-Os -pipe -mips32r2 -mtune=mips32r2 -fno-caller-saves
-SUBTARGETS:=generic nand
+SUBTARGETS:=generic linkduino nand
 
 LINUX_VERSION:=3.3.8
 
Index: target/linux/ar71xx/base-files.mk
===================================================================
--- target/linux/ar71xx/base-files.mk	(revision 34322)
+++ target/linux/ar71xx/base-files.mk	(working copy)
@@ -1,5 +1,11 @@
+ifeq ($(CONFIG_TARGET_ar71xx_linkduino),)
 define Package/base-files/install-target
 	rm -f $(1)/etc/config/network
 endef
-
-
+else
+define Package/base-files/install-target
+	rm -f $(1)/etc/uci-defaults/migrate-shadow
+	rm -f $(1)/etc/uci-defaults/vlan-migration
+	rm -f $(1)/etc/uci-defaults/wrt160nl
+endef
+endif
