--- davfs2-1.4.7.orig/src/webdav.c	2012-11-01 10:53:59.000000000 +0100
+++ davfs2-1.4.7/src/webdav.c	2012-11-01 12:35:23.000000000 +0100
@@ -406,10 +406,13 @@
             ne_ssl_certificate *server_cert
                     = ne_ssl_cert_read(args->servercert);
             if (!server_cert)
-                error(EXIT_FAILURE, 0, _("can't read server certificate %s"),
+                syslog(LOG_MAKEPRI(LOG_DAEMON, LOG_ERR),
+                       _("can't read server certificate %s"),
                       args->servercert);
-            ne_ssl_trust_cert(session, server_cert);
-            ne_ssl_cert_free(server_cert);
+            else {
+                ne_ssl_trust_cert(session, server_cert);
+                ne_ssl_cert_free(server_cert);
+            }
         }
 
         if (args->clicert) {
@@ -1969,13 +1972,24 @@
             printf(_("You only should accept this certificate, if you can\n"
                      "verify the fingerprint! The server might be faked\n"
                      "or there might be a man-in-the-middle-attack.\n"));
-            printf(_("Accept certificate for this session? [y,N] "));
+            printf(_("Accept certificate for this session? [y,N,w] "));
             char yesno[3];
             fflush(stdout);
             if (fgets(yesno, sizeof(yesno), stdin) == NULL)
                 abort();
             if (yesno[0] == 'y' || yesno[0] == 'Y')
                 ret = 0;
+            else if (yesno[0] == 'w' || yesno[0] == 'W') {
+                printf(_("Server certificate file name: "));
+                char filename[256];
+                fflush(stdout);
+                if (fgets(filename, sizeof(filename), stdin) == NULL)
+                    abort();
+                filename[strlen(filename) - 1] = '\0';
+                if (ne_ssl_cert_write(cert, filename))
+                    syslog(LOG_MAKEPRI(LOG_DAEMON, LOG_ERR),
+                           _("cannot write the server certificate"));
+            }
     } 
 
     if (failures & NE_SSL_NOTYETVALID)
diff -Naur davfs2-1.4.7.orig/src/webdav.c~ davfs2-1.4.7/src/webdav.c~
